### DISCUSSION TITLE
How to update current value by 1

### DISCUSSION THREAD

**Discussion by Marius555** [pocketbase/pocketbase#4369] [Q&A]
What is the correct way to update value. i have `Likes` in database set to 0, how to update this current value? do you have to make separate  request to check what is a current value or is it possible to increment value on Update request it self?

**Comment by ganigeorgiev** [comment]
The question is too vague and it is difficult to say exactly as it will depend on your specific use case and desired operation flow (eg. is everyone allowed to update the "likes" field or this is done programmatically triggered by some other action?).

In general, you can increment a `number` field value using the `+` modifier:

```js
await pb.collection("example").update("RECORD_ID", {
    "likes+": 1,
})
```

But note that the above doesn't run in a transaction because it will require db lock so if there are concurrent requests the counter may not be very reliable (aka. it is prone to race conditions).

Depending on your desired data structure and if you need details about the user "likes", an alternative approach cold be to use a separate collection that will store the liked post-user relations and a new "like" could be represented as a new entry in that collection (_you can enforce unique pair constraint from the "Unique constraints and indexes" collection settings_). To get the count of all likes per post/user/etc. you can create a View collection with your specific aggregation.

You can also check https://github.com/pocketbase/pocketbase/discussions/2235, https://github.com/pocketbase/pocketbase/discussions/4012.
